packages/turbo-gen/src/utils/plop.ts

```diff
@@ -1,27 +1,36 @@
 import fs from "fs-extra";
 import { Project } from "@turbo/workspaces";
 import nodePlop, { NodePlopAPI, PlopGenerator } from "node-plop";
+import { register } from "ts-node";
 import path from "path";
 import inquirer from "inquirer";
 import { searchUp, getTurboConfigs, logger } from "@turbo/utils";
 import { GeneratorError } from "./error";
 
-// TODO: Support a TS config file
-const TURBO_GENERATOR_CONFIG = path.join("turbo", "generators", "config.js");
+const SUPPORTED_CONFIG_EXTENSIONS = ["ts", "js", "cjs"];
+const TURBO_GENERATOR_DIRECTORY = path.join("turbo", "generators");
 
-// support root plopfile so that users with existing configurations can use them immediately
-const DEFAULT_ROOT_CONFIG_LOCATIONS = [
-  TURBO_GENERATOR_CONFIG,
-  "plopfile.js",
-  "plopfile.cjs",
-  "plopfile.mjs",
+// config formats that will be automatically loaded from within workspaces
+const SUPPORTED_WORKSPACE_GENERATOR_CONFIGS = SUPPORTED_CONFIG_EXTENSIONS.map(
+  (ext) => path.join(TURBO_GENERATOR_DIRECTORY, `config.${ext}`)
+);
+
+// config formats that will be automatically loaded from the root (support plopfiles so that users with existing configurations can use them immediately)
+const SUPPORTED_ROOT_GENERATOR_CONFIGS = [
+  ...SUPPORTED_WORKSPACE_GENERATOR_CONFIGS,
+  ...SUPPORTED_CONFIG_EXTENSIONS.map((ext) => path.join(`plopfile.${ext}`)),
 ];
 
 export type Generator = PlopGenerator & {
   basePath: string;
   name: string;
 };
 
+// init ts-node for plop to support ts configs
+register({
+  transpileOnly: true,
+});
```

Comment:
Plop does support TS configs, but it relies on ts-node being configured https://github.com/plopjs/plop/issues/192#issuecomment-567129666